client/src/listings/CreateListing.js

```diff
@@ -69,39 +80,179 @@ export default function CreateListing() {
           </Grid>
         </Grid>
         <Divider />
-        <Grid container justify='space-between'>
-          <Grid item xs={5}>
+        <Grid container justify='space-between' spacing={2}>
+          <Grid item xs={6}>
             <TextField name='title' required fullWidth label='Title' autoFocus onChange={handleInputChange} />
           </Grid>
           {sessionStorage.getItem('accountType') === '1' ? (
             <Grid item xs={5}>
-              <FormControl fullWidth required onChange={handleInputChange}>
-                <InputLabel htmlFor='categoryId'>Category</InputLabel>
-                <Select name='categoryId' native defaultValue='' id='categoryId'>
+            <FormControl fullWidth required onChange={handleInputChange}>
+              <InputLabel htmlFor='categoryId'>Category</InputLabel>
+              <Select name='categoryId' native defaultValue='' id='categoryId'>
+                <option value='' />
+                {categoryOptions.length > 0 && Object.keys(subcategoryOptions).length > 0 ? (
+                  categoryOptions.map((cat) => (
+                    <optgroup key={cat.id} label={`${cat.name}`}>
+                      {subcategoryOptions[cat.id] ? (
+                        subcategoryOptions[cat.id].map((subcat) => (
+                          <option key={subcat.id} value={subcat.id}>{`${subcat.name}`}</option>
+                        ))
+                      ) : (
+                        <></>
+                      )}
+                    </optgroup>
+                  ))
+                ) : (
                   <option value='' />
-                  {categoryOptions.length > 0 && Object.keys(subcategoryOptions).length > 0 ? (
-                    categoryOptions.map((cat) => (
-                      <optgroup key={cat.id} label={`${cat.name}`}>
-                        {subcategoryOptions[cat.id] ? (
-                          subcategoryOptions[cat.id].map((subcat) => (
-                            <option key={subcat.id} value={subcat.id}>{`${subcat.name}`}</option>
-                          ))
-                        ) : (
-                          <></>
-                        )}
-                      </optgroup>
-                    ))
-                  ) : (
-                    <option value='' />
-                  )}
-                </Select>
-              </FormControl>
+                )}
+              </Select>
+            </FormControl>
+          </Grid>
+          ) : ( listing.categoryId='664' )}
+          <Grid item xs={6}>
+            <TextField name='subDescription' required fullWidth label='Short Description' onChange={handleInputChange} />
+          </Grid>
+        </Grid>
+        <br />
+        {sessionStorage.getItem('accountType') === '1' ? (
+            <Grid container xs={12}>
+              <br/>
+              {['104', '114', '124', '134', '144', '154', '164', '174'].includes(listing.categoryId) ? (
```

Comment:
These should use the ids loaded from the DB, as discussed. You can use the subcategories state variable to look up the subcategory ids for a given parent category, then change this code to check if the current categoryId matches any of the subcategory ids in the given parent(s). You can print out the values of the subcategories state to see how that's exactly stored.